lib/rules/lint-template-length.js

```diff
@@ -0,0 +1,93 @@
+'use strict';
+
+/*
+ Enforce template size constraints
+
+ The following values are valid configuration:
+
+   * boolean -- `true` for enabled (same as `unix`) / `false` for disabled
+   * object --
+     * max {number} - the longest a template should be without failing a test (assuming the
+       right thing to do would be to split the template up into pieces.
+     * min {number} - the shortest a template should be without failing a test (assuming the
+       right thing to do would be to inline the template.
+ */
+
+const Rule = require('./base');
+
+const MAX_LENGTH = 100;
+const MIN_LENGTH = 5;
+
+const DEFAULT_CONFIG = {
+  max: MAX_LENGTH,
+  min: MIN_LENGTH,
+};
+
+function isValidConfigObjectFormat(config) {
+  for (let key in config) {
+    let value = config[key];
+    let valueIsInteger = Number.isInteger(value);
+
+    if (key === 'min' && !valueIsInteger) {
+      return false;
+    } else if (key === 'max' && !valueIsInteger) {
+      return false;
+    }
+  }
+  return true;
+}
+
+module.exports = class LengthValidation extends Rule {
+  parseConfig(config) {
+    let configType = typeof config;
+
+    let errorMessage = 'The template-length rule accepts one of the following values.\n ' +
+      '  * boolean - `true` to enable / `false` to disable\n' +
+      '  * object -- An object with the following keys:' +
+      '    * `min` -- Minimum length of a template ' +
+      '    * `max` -- Maximum length of a template. ' +
+      '\nYou specified `' + JSON.stringify(config) + '`';
+
+    switch (configType) {
+    case 'boolean':
+      // if `true` use `DEFAULT_CONFIG`
+      return config ? DEFAULT_CONFIG : {};
+    case 'object':
+      if (isValidConfigObjectFormat(config)) {
+        return config;
+      } else {
+        throw new Error(errorMessage);
+      }
+    case 'undefined':
+      return {};
+    default:
+      throw new Error(errorMessage);
+    }
+  }
+
+  visitor() {
+    return {
+      Program(node) {
```

Comment:
The way the comment disabling system works is that when it visits a comment node which contains an template lint directive, it updates the current configuration such that the rule is disabled. When the rule is disabled, none of the other visitors are called.

The "rule can be disabled via comment" tests are failing because of the way this visitor is structured. Specifically, it is not possible for the visitor system to "see" a comment node that disables the rule before the `Program` visitor here is invoked (which would immediately call `this.log`).  Thankfully, the fix is relatively straightforward though:

```js
visitor() {
  return {
    Program: {
      exit(node) {
        // move your implementation here, since it is in the exit handler it will no longer
        // be called if the rule has been disabled by any inline comments within the file
      }
    }
  }
}
```